.github/workflows/tests.yml

```diff
@@ -0,0 +1,78 @@
+# This workflow builds and runs test
+
+name: tests
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "develop"
+    paths-ignore:
+      - "docs/**"
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        py-version:
+          - 3.6
+          - 3.7
+          - 3.8
+        mpi:
+          - openmpi
+        os:
+          - ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install MPI (${{ matrix.mpi }})
+        run: bash .bash/install-${{ matrix.mpi }}.sh
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Install via pip
+        run: |
+          sudo pip install -e .[all]
+
+      - name: Run serial tests
+        run: |
+          pytest -v
+
+      - name: Run parallel tests (procs=2)
+        run: |
+          mpirun -n 2 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=4)
+        run: |
+          mpirun -n 4 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=7)
+        run: |
+          mpirun -n 7 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run parallel tests (procs=12)
+        run: |
+          mpirun -n 12 --oversubscribe python -m pytest test/parallel_tests/ --with-mpi
+
+      - name: Run tests with coverage
```

Comment:
That should generate xml coverage report for serial and parallel tests although not sure what should happen after export statement i.e update the badge etc

Reply:
Also, that runs for three python versions. Might be better to add it to different workflow?

Reply:
Better to test more than less. Maybe there are too many MPI cases, but keep it like this for now. 
Actually instead of duplicating the code, would be nice to have it as a strategy `nprocs=[1,2,4,7,12]`, where nprocs=1 is run without MPI.

Reply:
Sorry I take it back because in this case would require to install everything multiple times.
Maybe you can put strategies in a step? try to check!

Reply:
I'll try that, you're right.  For the coverage report i.e lines 71-78 below though do we need to update the badge too once this is run or not needed? If so could you please advice on how to do that?

Reply:
The update of the badge is done it in the deployment. It could be done also here, but we will discuss it next time.
FYI: https://github.com/Lyncs-API/lyncs.io/blob/9f37c43cd0802d6df9468fec120cef869db010a9/.github/workflows/ci_cd.yml#L34